projects/robots/boston_dynamics/spot/protos/SpotRightLeg.proto

```diff
@@ -0,0 +1,841 @@
+#VRML_SIM R2020a utf8
+# license: Apache License 2.0.
+# license url: http://www.apache.org/licenses/LICENSE-2.0.
+# documentation url: https://www.cyberbotics.com/doc/guide/spot
+# The "Spot" is a dog robot developed by Boston Dynamics.
+# This PROTO is the definition of right legs.
+# Created by Greg McKechnie under Common Creative license: https://sketchfab.com/3d-models/spot-mini-rigged-5dcbee77730640269cef5bd2587e328a
+
+PROTO SpotRightLeg [
+  field SFVec3f     translation  0 0 0        # Is `Transform.translation`.
+  field SFRotation  rotation     0 1 0 0      # Is `Transform.rotation`.
+  field SFString    name         "right leg"  # Is `Solid.name`.
+]
+{
+%{
+  local legName = fields.name.value
+}%
+  DEF SHOULDER_ELEVATION_JOINT HingeJoint {
+    jointParameters HingeJointParameters {
+      axis 0 0 1
+      anchor 0.05283 1.662418 -0.513
+      minStop -0.6
+      maxStop 0.5
+    }
+    device [
+      PositionSensor {
+        name %{= "\"" .. legName .. " shoulder elevation sensor\"" }%
+      }
+      RotationalMotor {
+        name %{= "\"" .. legName .. " shoulder elevation motor\"" }%
+        maxTorque 50
+      }
+    ]
+    endPoint DEF SHOULDER Solid {
+      translation IS translation
+      rotation IS rotation
+      children [
+        DEF SHOULDER_ROTATION_JOINT HingeJoint {
+          jointParameters HingeJointParameters {
+            anchor 0 1.66177 -0.82329
+            minStop -1
+            maxStop 1.7
+          }
+          device [
+            PositionSensor {
+              name %{= "\"" .. legName .. " shoulder rotation sensor\"" }%
+            }
+            RotationalMotor {
+              name %{= "\"" .. legName .. " shoulder rotation motor\"" }%
+              maxTorque 50
+            }
+          ]
+          endPoint DEF UPPERARM Solid {
+            children [
+              DEF TODO SliderJoint {
+                jointParameters JointParameters {
+                }
+                device [
+                  LinearMotor {
+                  }
```

Comment:
If I understand correctly this is a kind of 'internal' motor which is linked to the elbow right?
In any case it should have a name, as currently there is a name clash because this motor have 4 times the same name (because of the 4 legs).

Reply:
That's right. If you look at the real robot, you can see that it uses a kind of piston to raise and lower the forearm of the leg.